File: SoccerAgent/src/test/java/onflows/worldmodel/Relation.java
package onflows.worldmodel;

/**
 * Created by Sergey on 13.09.2016.
 */
public class Relation<D, R> {
    private D domain;
    private R range;


    public D getDomain() {
        return domain;
    }

    public void setDomain(D domain) {
        this.domain = domain;
    }

    public R getRange() {
        return range;
    }

    public void setRange(R range) {
        this.range = range;
    }
}

File: ProtoSASOnScala/src/main/scala/dd/sas/pipeline/worldmodel/Relation.scala
package dd.sas.pipeline.worldmodel

/**
  * Created by Sergey on 24.09.2016.
  */
class Relation[D, R](val domain: D, val range: R) {
  def getDomain: D = domain

  def getRange: R = range

  override def toString() = domain + " " + range
}
